### PR TITLE
Fix/use sqlops engine version

### PR DESCRIPTION
**Pull request by llali**
sqlops was checking the "vscode" version of the extensions for compatibility with engine. Changed the code to use "engine.sqlops" as the engine version and setting it to * as default so it won't break for the extensions that don't have the field

### CODE REVIEW

File: src/vs/platform/extensionManagement/common/extensionManagement.ts [hunk 1]
```diff
@@ -105,7 +105,7 @@ export interface IExtensionManifest {
 	name: string;
 	publisher: string;
 	version: string;
-	engines: { vscode: string };
+	engines: { vscode: string; sqlops?: string };
```
Comment (MattIrv) [inline src/vs/platform/extensionManagement/common/extensionManagement.ts]: We should mark this and the changes in the file below as `// {{SQL CARBON EDIT}}`

File: src/vs/platform/extensions/node/extensionValidator.ts [hunk 2]
```diff
@@ -219,7 +221,8 @@ export function isValidExtensionVersion(version: string, extensionDesc: IReduced
 		return true;
 	}
 
-	return isVersionValid(version, extensionDesc.engines.vscode, notices);
+	// {{SQL CARBON EDIT}}
+	return (extensionDesc.engines.sqlops && extensionDesc.engines.sqlops === '*') || isVersionValid(version, extensionDesc.engines.vscode, notices);
```
Comment (MattIrv) [inline src/vs/platform/extensions/node/extensionValidator.ts]: Did you mean to leave `extensionDesk.engines.vscode` in the version check here? We want to check the sqlops version, right?

Comment (llali) [reply src/vs/platform/extensions/node/extensionValidator.ts]: no changed it to sqlops 

---
In reply to: [177604383](https://github.com/Microsoft/sqlopsstudio/pull/1011#discussion_r177604383) [](ancestors = 177604383)

**Comment by anthonydresser** [comment]
Will this fix 
image

**Pull request merged**